libs/store/src/lib/effects/race.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  AppPlayerId,
+  AppPlayerProfiles,
+  AppRace,
+  AppRaceId,
+  AppTournament,
+  AppTournamentId,
+  AppTournamentState,
+} from '@razor/models';
+import {
+  calculateTimeoutTimer,
+  extractId,
+  extractIdType,
+  generateLeaderboard,
+  giveZeroPadding,
+} from '@razor/util';
+import { playerNotFound, tournamentNotFound } from '../loggers';
+import { raceNotFound } from '../loggers/race';
+import {
+  endCountdownPayload,
+  endRacePayload,
+  startCountdownPayload,
+} from '../payloads';
+import { Dispatch, RootState } from '../store';
+
+//TODO: should move this to server later
+const loadRacingText = async (): Promise<string> => {
+  const url = 'http://www.metaphorpsum.com/paragraphs/1/8';
+
+  return fetch(url)
+    .then(response => response.text())
+    .then(data => {
+      return data;
+    });
+};
+
+export const startCountdown = async (
+  dispatch: Dispatch,
+  payload: startCountdownPayload,
+  state: RootState,
+): Promise<void> => {
+  const {
+    tournamentId,
+    playerId,
+  }: { tournamentId: AppTournamentId; playerId: AppPlayerId } = payload;
+
+  if (!state.game.tournamentsModel[tournamentId])
+    tournamentNotFound(dispatch, tournamentId, `Started by: ${playerId}`);
+  if (!state.game.playersModel[playerId])
+    playerNotFound(
+      dispatch,
+      playerId,
+      `While tournament starting: ${tournamentId}`,
+    );
+
+  const numberOfRacesBefore =
+    state.game.tournamentsModel[tournamentId] &&
+    state.game.tournamentsModel[tournamentId].raceIds.length;
+  const raceIndex = numberOfRacesBefore
+    ? giveZeroPadding(numberOfRacesBefore.toString(), 3)
+    : '000';
+  const raceId: AppRaceId = `${tournamentId}-R:${raceIndex}`;
+  const players: AppPlayerProfiles = {};
+
+  for (const id of state.game.tournamentsModel[tournamentId].playerIds) {
+    const player = state.game.playersModel[id];
+    if (!player) {
+      if (!state.game.playersModel[playerId])
+        playerNotFound(
+          dispatch,
+          playerId,
+          `While players are being added to: ${tournamentId}`,
+        );
+    } else {
+      players[id] = {
+        name: player.name,
+        avatarLink: player.avatarLink,
+      };
+    }
+  }
+
+  const raceText = await loadRacingText();
```

Comment:
I think that the racingText must come as an input payload for this effect; because, including the text loading logic here couples our game with the text provider.

The text provider would ideally live in the backend; having no connection to the game state, or its definitions.
WDYT?

Reply:
Yes, this was changed in PR #14 under commit 4cc6db0746f129bb874452203467d069dfa797e0 and a89f04352fcf13545275e01bbd41b93a1aaa26ac